### PR TITLE
js: Add Placement config to Stream

### PR DESCRIPTION
**Pull request by wallyqs**
Allows being able to define the cluster and tags on stream create.

```go
js.AddStream(&nats.StreamConfig{
	Name: "TEST",
	Placement: &nats.Placement{
		Cluster: cluster,
		Tags:    []string{"NODE_0"},
		},
	})
```

### CODE REVIEW

**Comment by ripienaar** [comment]
Just a note atm server doesn’t do anything with tags. Will do before GA hopefully

File: jsm.go [hunk 1]
```diff
@@ -64,20 +64,27 @@ type JetStreamManager interface {
...
+	MaxBytes     int64            `json:"max_bytes"`
+	Discard      DiscardPolicy    `json:"discard"`
+	MaxAge       time.Duration    `json:"max_age"`
+	MaxMsgSize   int32            `json:"max_msg_size,omitempty"`
+	Storage      StorageType      `json:"storage"`
+	Replicas     int              `json:"num_replicas"`
+	NoAck        bool             `json:"no_ack,omitempty"`
+	Template     string           `json:"template_owner,omitempty"`
+	Duplicates   time.Duration    `json:"duplicate_window,omitempty"`
+	Placement    *StreamPlacement `json:"placement,omitempty"`
```
Comment (derekcollison) [inline jsm.go:81]: Would just call it placement.

**Review by derekcollison — APPROVED** [review]
LGTM - I may just call it placement, but ok with StreamPlacement too.

File: jsm.go (hunk 1 above)
Comment (wallyqs) [reply jsm.go:81]: ok will update to Placement

**Review by variadico — APPROVED** [review]

**Pull request merged**